pkg/kubectl/cmd/util/helpers.go

```diff
@@ -620,7 +620,7 @@ func AddInclude3rdPartyVarFlags(cmd *cobra.Command, include3rdParty *bool) {
 func GetResourcesAndPairs(args []string, pairType string) (resources []string, pairArgs []string, err error) {
 	foundPair := false
 	for _, s := range args {
-		nonResource := strings.Contains(s, "=") || strings.HasSuffix(s, "-")
+		nonResource := strings.Contains(s, "=") || (strings.HasSuffix(s, "-") && s != "-")
```

Comment:
If you're fixing this for hyphen (`-`) can you also add similar to check for the equal sign (`=`). I'm guessing the same error will appear if you use it.